### PR TITLE
ARROW-13756: [Python] Error in pandas conversion for datetimetz column index

### PR DESCRIPTION
**Pull request by AlenkaF**
Specifically deal with timezone-aware data in the code to reconstruct the column Index `(_reconstruct_columns_from_metadata)`.

### CODE REVIEW

File: python/pyarrow/tests/test_pandas.py [hunk 1]
```diff
@@ -237,6 +237,17 @@ def test_multiindex_columns_with_dtypes(self):
         df = pd.DataFrame([(1, 'a'), (2, 'b'), (3, 'c')], columns=columns)
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_multiindex_columns_with_tz(self):
```
Comment (jorisvandenbossche) [inline python/pyarrow/tests/test_pandas.py:240]: Small note on the test name: this is not actually using a MultiIndex column (it's a single level, but using a DatetimeIndex for the column names). So I would rename the test and move it up a little below where there are tests for different column index types (there seems to be already a test related to this)

**Review by jorisvandenbossche — COMMENTED** [review]
Some small comments

**Comment by AlenkaF** [comment]
@jorisvandenbossche this is ready for another round of review. Thanks!

**Review by jorisvandenbossche — APPROVED** [review]
Sorry for the slow follow-up here. But looks good to me now! Can you maybe just do a rebase to ensure it's still working fine on top of latest master?

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request closed without merging**